### PR TITLE
DOCSP-33246 Convert Views Content

### PR DESCRIPTION
**Pull request by ianf-mongodb**
## DESCRIPTION

Task page, how to convert SQL views to MongoDB code.

## STAGING

https://preview-mongodbianfmongodb.gatsbyjs.io/docs-relational-migrator/DOCSP-33246/code-generation/query-converter/convert-views/

## JIRA

https://jira.mongodb.org/browse/DOCSP-33246

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=654e990e5b686d6bd9347a41

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/code-generation/query-converter/convert-views.txt [hunk 1]
```diff
@@ -5,7 +5,193 @@ Convert Views
...
+  .. step:: Select stored views
+
+     a. On the :guilabel:`Import Database Objects` modal, click
+        :guilabel:`Database`.
+     #. Click the :icon-fa5:`chevron-right` icon next to your schema.
+     #. Click the :icon-fa5:`chevron-right` icon next to
+        :guilabel:`Views`.
+     #. You can toggle views for conversion by clicking
+        the :icon-fa5:`check-square` icon next to a view's
+        name. All views are selected for conversion by default.
```
Comment (jeff-allen-mongo) [inline source/code-generation/query-converter/convert-views.txt:63]: Can we make this step more active voice?

To select the views to convert, click the (...). By default, all views are selected.

File: source/code-generation/query-converter/convert-views.txt [hunk 2]
```diff
@@ -5,7 +5,193 @@ Convert Views
...
+     #. Click :guilabel:`Save`.
+
+  .. step:: Convert and test code
+
+     a. Click a view's name from the left-hand pane under
+        :guilabel:`Views`.
+        
+        The SQL code is loaded into the :guilabel:`Imported View`
```
Comment (jeff-allen-mongo) [inline source/code-generation/query-converter/convert-views.txt:71]:
```suggestion
        The SQL code is displayed in the :guilabel:`Imported View`
```

File: source/code-generation/query-converter/convert-views.txt [hunk 3]
```diff
@@ -5,7 +5,193 @@ Convert Views
...
+        pane.
+     #. Click the :guilabel:`Convert` button. Wait for the query 
+        converter to convert your code. 
+        
+        The converted MongoDB code 
+        displays in the :guilabel:`Converted MongoDB Query` pane.
```
Comment (jeff-allen-mongo) [inline source/code-generation/query-converter/convert-views.txt:77]:
```suggestion
        is displayed in the :guilabel:`Converted MongoDB Query` pane.
```

File: source/code-generation/query-converter/convert-views.txt [hunk 4]
```diff
@@ -5,7 +5,193 @@ Convert Views
...
+     #. Test and validate the generated MongoDB code.
+
+     .. tip::
+
+        .. include:: /includes/fact-query-converter-filter.rst
+
+Examples
+--------
+
+The query converter can convert your SQL views to MongoDB code.
```
Comment (jeff-allen-mongo) [inline source/code-generation/query-converter/convert-views.txt:90]: I don't think we need this initial sentence.

File: source/code-generation/query-converter/convert-views.txt [hunk 5]
```diff
@@ -5,7 +5,193 @@ Convert Views
...
+
+     .. tip::
+
+        .. include:: /includes/fact-query-converter-filter.rst
+
+Examples
+--------
+
+The query converter can convert your SQL views to MongoDB code. 
+The examples below show PostGreSQL views from the
```
Comment (jeff-allen-mongo) [inline source/code-generation/query-converter/convert-views.txt:91]:
```suggestion
The examples below show PostgreSQL views from the
```

File: source/code-generation/query-converter/convert-views.txt [hunk 6]
```diff
@@ -5,7 +5,193 @@ Convert Views
...
+:ref:`example Northwind database<create-project-sample-schema>` 
+converted to MongoDB code.
+
+The following example shows a ``inner join`` view converted to MongoDB:
```
Comment (jeff-allen-mongo) [inline source/code-generation/query-converter/convert-views.txt:95]:
```suggestion
The following example shows an ``inner join`` view converted to MongoDB:
```

Comment (jeff-allen-mongo) [inline source/code-generation/query-converter/convert-views.txt:95]: [question]

Does "inner join" need to be monospaced?

File: source/code-generation/query-converter/convert-views.txt [hunk 7]
```diff
@@ -5,7 +5,193 @@ Convert Views
...
+
+.. tabs::
```
Comment (jeff-allen-mongo) [inline source/code-generation/query-converter/convert-views.txt:97]: I'm not sure the tabs are particularly helpful here. I think it's helpful to be able to see the input and output at once. Tabs would make sense if we were maybe showing different types of relational databases (in addition to Postgres), but I think in this case we might want to show both code-blocks together.

**Review by jeff-allen-mongo — CHANGES_REQUESTED** [review]
@ianf-mongodb thanks for these updates. I left some suggestions for you to consider.

Jeff

File: source/code-generation/query-converter/convert-views.txt [hunk 8]
```diff
@@ -5,7 +5,193 @@ Convert Views
...
+
+Before you Begin
+----------------
+
+.. include:: /includes/fact-query-converter-disclaimer.rst
+
+Steps
+-----
+.. procedure::
+  :style: normal
```
Comment (jeff-allen-mongo) [inline source/code-generation/query-converter/convert-views.txt:30]: Vale raised a false positive for this line. We created https://jira.mongodb.org/browse/DOCSP-34343 to investigate.

**Review by ianf-mongodb — COMMENTED** [review]
Thanks @jeff-allen-mongo, this has been updated and is ready for another look.

**Review by jeff-allen-mongo — APPROVED** [review]
LGTM

**Comment by ianf-mongodb** [comment]
@tomhollander This is ready for review!

**Review by tomhollander — CHANGES_REQUESTED** [review]
Thanks, content looks good overall. However I think we need a bit of extra information to help people use the output.

The query converter takes the SQL query used to create the view and converts it to the equivalent MQL. If the user wants to create a MongoDB View from the query, they will need to take the pipeline from the `aggregate` command and pass it to the `createView` shell command - it would be great to have an example of this.

Alternatively, users may want to use the query in their code without creating a MongoDB view, in which case they can keep it as-is.

**Review by ianf-mongodb — COMMENTED** [review]
Hi @tomhollander, I added the additional createView code for each example on the page. The formatting on the code blocks might need some adjusting, but wanted to get your thoughts on the presentation as that can be cleaned up before publish.

**Review by tomhollander — APPROVED** [review]
Thanks for the updates!

**Pull request merged**